src/jit/lsra.cpp

```diff
@@ -8581,20 +8570,16 @@ void LinearScan::insertMove(
     // This var can't be marked lvRegister now
     varDsc->lvRegNum = REG_STK;
 
-    var_types lclTyp = varDsc->TypeGet();
-    if (varDsc->lvNormalizeOnStore())
-    {
-        lclTyp = genActualType(lclTyp);
-    }
+    var_types lclTyp = genActualType(varDsc->TypeGet());
```

Comment:
We have the following cases

1) lcl-var is getting spilled from reg to stack:
use type of VarDsc on src.
Spill logic will use the actual/normalized type of src while writing reg to stack.

2) lcl-var is getting loaded from stack to reg:
use the type of VarDsc on src.  Codgen for GT_LCL_VAR will sign/zero-extend  the result while loading from stack to reg.

3) lcl-var is getting copied from one reg to another reg:
Use genActualType() on top and src, because codegen will use mov while generating code for GT_COPY.


---
In reply to: [82099257](https://github.com/dotnet/coreclr/pull/7504#discussion_r82099257) [](ancestors = 82099257)